mobile/ios/device/ios-device.ts

```diff
@@ -339,16 +340,22 @@ export class IOSDevice implements Mobile.IiOSDevice {
 	}
 
 	public openDeviceLogStream() {
-		if(this.deviceInfo.status !== constants.UNREACHABLE_STATUS) {
-			let iOSSystemLog = this.$injector.resolve(iOSProxyServices.IOSSyslog, {device: this});
-			iOSSystemLog.read();
+		if (this.deviceInfo.status !== constants.UNREACHABLE_STATUS) {
+			this.iOSSystemLog = this.$injector.resolve(iOSProxyServices.IOSSyslog, { device: this });
+			this.iOSSystemLog.read();
+		}
+	}
+
+	public closeDeviceLogStream() {
+		if (this.iOSSystemLog) {
+			this.iOSSystemLog.close();
```

Comment:
You can assign to `null` here too in order to release the pointer to this object